bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/threading/ThreadsafeTimers.java

```diff
@@ -95,8 +141,11 @@ public long setTimeout(Runnable callback, Long delay) {
      */
     public long setTimeout(Runnable callback, Long delay, Object... args) {
         long id = lastId.incrementAndGet();
-        ScheduledCompletableFuture<Object> future = createFuture(id, ZonedDateTime.now().plusNanos(delay * 1000000),
-                callback);
+        ScheduledCompletableFuture<Object> future = scheduler.schedule(() -> {
+            synchronized (lock) {
+                callback.run();
```

Comment:
Should we remove the future from the map here ?  so `idSchedulerMapping.remove(id)` ?  I noticed the only time we remove references to futures is when we completely clear the map at init/reload time?

Reply:
It's also cleared on `clearTimeout`, but I agree that it should be cleared here too.

Reply:
I agree, I'll change this.